lib/tasks/scripts/message_patients.rb

```diff
@@ -41,16 +41,26 @@ def send_messages
       phone_number = phone_number_for(patient)
 
       next unless phone_number
-      notification_service = NotificationService.new
-      if whatsapp?
-        notification_service.send_whatsapp(phone_number, message)
-      elsif sms?
-        notification_service.send_sms(phone_number, message)
-      end
-      if notification_service.failed?
+      notification_service = TwilioApiService.new
+      context = {
+        calling_class: self.class.name,
+        patient_id: patient.id,
+        communication_type: channel
+      }
+
+      begin
+        response = if whatsapp?
+          notification_service.send_whatsapp(recipient_number: phone_number, message: message, context: context)
+        elsif sms?
+          notification_service.send_sms(recipient_number: phone_number, message: message, context: context)
+        end
+      rescue TwilioApiService::Error
```

Comment:
Ah I see. In this script, we don't want to halt on exception, so we rescue and report. Neat.